### PR TITLE
Smaraba flask

### PR DESCRIPTION
**Pull request by smarabattula**
**Major updates:**
- Added Flask backend implementation into the file "ui1.py"
- Added Basic Frontend with HTML and JavaScript part to upload and send requests to server
- Added status update functionality using AJAX (Async Javascript)

**Fixed minor bugs:**
- Slight Documentation updates
- Some technical issues during cloning repo
  - The library "appscript" isn't compatible in WindowsOS
  - There is a file name called * that won't unzip as WindowsOS doesn't support * in filenames while Linux does
 - Folder reorganization to save uploaded files and anki files in directories
 - Code cleanup
   - Unused libraries in code from previous versions
   - Some libraries like openAI has updated api functions had to reimplement with corrected function calls (ex: openai.ChatCompletion.create())

### CODE REVIEW

File: code/extract_sizes.py [hunk 1]
```diff
@@ -1,45 +1,46 @@
...
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
 """
 
-File completing step 2: given a pdf document, return a dictionary
-of headers and paragraphs
+File completing step 2: given a pdf document
+return a dictionaryof headers and paragraphs
```
Comment (MatthewSills) [inline code/extract_sizes.py:26]: Missing space between dictionary and of.

File: code/ui.py [hunk 2]
```diff
@@ -39,7 +39,7 @@
 # import filedialog module
 
 
-def process_(file):  # , progress_callback, finish_callback):
+def process_(file,):  # , progress_callback, finish_callback):
```
Comment (MatthewSills) [inline code/ui.py:42]: Is there supposed to be a comma after file here?

**Review by MatthewSills — APPROVED** [review]
I made some comments where I thought there might be typos. I cannot comment on the flask implementation itself as I have not used flask before but nothing stood out as obviously wrong.

File: code/extract_sizes.py (hunk 1 above)
Comment (smarabattula) [reply code/extract_sizes.py:26]: Can be fixed in coming pull requests

File: code/ui.py (hunk 2 above)
Comment (smarabattula) [reply code/ui.py:42]: It shouldn't be, but it doesn't matter as Python function syntax will allow it. 
@varundeepakgudhe can fix it.

**Pull request merged**